src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -1,56 +1,117 @@
 import { h, Component} from 'preact'
+import classNames from 'classnames'
+import io from 'socket.io-client'
+
 import theme from '../../Theme/style.css'
 import style from './style.css'
-import classNames from 'classnames'
+import Spinner from '../../Spinner'
 import { trackComponent } from '../../../Tracker'
+import { versionToBase36 } from '../../utils/versionMap'
 
 class CrossDeviceLink extends Component {
   constructor(props) {
-    super(props);
-    this.state = { copySuccess: false }
+    super(props)
+    this.state = {roomId: null}
+
+    if (!props.socket) {
+      const socket = io(process.env.DESKTOP_SYNC_URL)
+      props.actions.setSocket(socket)
+      this.listen(socket)
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.socket) {
```

Comment:
which scenario does `else` happen?

Reply:
The very first time you create the component the socket is not in the store so it's not in the props.

I agree this feels odd - there may be a case where we end up calling listen twice. Perhaps we should track if we have called listen with some state

Reply:
it's fine to have it here, if you have the listener set in `propsWillChange`

Reply:
I don't understand - can we discuss?